### PR TITLE
Add some documentation about traffic features.

### PR DESCRIPTION
**Pull request by t4lz**
https://github.com/metalbear-co/mirrord/issues/468

If the stuff I wrote is correct I think it's important for users to understand.
What do you think?

### CODE REVIEW

File: content/docs/reference/traffic/index.md [hunk 1]
```diff
@@ -18,8 +18,11 @@ mirrord lets users debug incoming network traffic by mirroring or stealing the t
 
 #### Mirroring
 
-mirrord's default configuration is to the mirror traffic that reaches the remote pod, i.e. run the local process in the context of cloud environment without
-disrupting incoming traffic.
+mirrord's default configuration is to mirror incoming TCP traffic from the remote pod's incoming connections, i.e.
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:21]:
```suggestion
mirrord's default configuration is to mirror incoming TCP traffic from the remote pod, i.e.
```

File: content/docs/reference/traffic/index.md [hunk 2]
```diff
@@ -18,8 +18,11 @@ mirrord lets users debug incoming network traffic by mirroring or stealing the t
...
+run the local process in the context of cloud environment without disrupting incoming traffic for the remote pod.
+Any responses by the local process to requests on incoming connections (all outgoing TCP packets on sessions not
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:23]:
```suggestion
Any responses by the local process to the mirrored requests are dropped, and so whatever application is running on the remote pod continues to operate normally while the traffic is mirrored to the local process.```

File: content/docs/reference/traffic/index.md [hunk 3]
```diff
@@ -165,9 +168,17 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
 
 ## Outgoing
 
-mirrord's outgoing traffic feature intercepts outgoing requests from the local process and sends them through the remote pod instead.
-Responses are then routed back to the local process. A simple use case of this feature is when a user wants their local process to make
-an API call to another service in the k8s cluster, for example, a database read/write.
+mirrord's outgoing traffic feature intercepts outgoing requests (in outgoing connections) from the local process and
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:171]:
```suggestion
mirrord's outgoing traffic feature intercepts outgoing requests from the local process and 
```

File: content/docs/reference/traffic/index.md [hunk 4]
```diff
@@ -165,9 +168,17 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
 
 ## Outgoing
 
-mirrord's outgoing traffic feature intercepts outgoing requests from the local process and sends them through the remote pod instead.
-Responses are then routed back to the local process. A simple use case of this feature is when a user wants their local process to make
-an API call to another service in the k8s cluster, for example, a database read/write.
+mirrord's outgoing traffic feature intercepts outgoing requests (in outgoing connections) from the local process and 
+sends them through the remote pod instead. Responses are then routed back to the local process.
+A simple use case of this feature is enabling the local process to make an API call to another service in the k8s
+cluster, for example, a database read/write. Note that in such an example, if a user request triggers an outgoing API
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:174]: The note here should be a note, not part of the introductory paragraph.

Comment (eyalb181) [reply content/docs/reference/traffic/index.md:174]: Also the scenario described here is unclear - what does it mean for a user request to trigger an outgoing API call over the network? Who's the recipient of the request? What does 'over the network' mean?

File: content/docs/reference/traffic/index.md [hunk 5]
```diff
@@ -165,9 +168,17 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+call over the network, and [traffic stealing](#stealing) is not enabled, this could lead to two calls being sent out
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:175]: Note that stealing/mirroring are mutually exclusive - stealing isn't something you enable on top of mirroring. I would rephrase along the lines of - "if mirrord is configured to mirror rather than steal incoming traffic".

File: content/docs/reference/traffic/index.md [hunk 6]
```diff
@@ -165,9 +168,17 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+of the remote pod for one user request - one API call generated by the remote process, and one generated by the local
+process (and emitted out of the remote pod with the help of mirrord).
+Outgoing UDP is currently only intercepted and forwarded by mirrord if the application binds a non-0 port and makes
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:178]:
```suggestion
For UDP, outgoing traffic is currently only intercepted and forwarded by mirrord if the application binds a non-0 port and makes
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: content/docs/reference/traffic/index.md (hunk 4 above)
Comment (t4lz) [reply content/docs/reference/traffic/index.md]: Is the note clearer now? Do you want me to just delete the note?

Comment (t4lz) [reply content/docs/reference/traffic/index.md]: Btw, the other notes on this website are not all uniform, and this note is a mix of the styles (bold + quote). This is what it looks like: 
image
Please feel free to let me know which style you prefer for this note and whether you would like me to also change other notes in this website to the chosen style.

Comment (eyalb181) [reply content/docs/reference/traffic/index.md]: Can you point me to the other notes? I couldn't find any.

Comment (t4lz) [reply content/docs/reference/traffic/index.md]: "Note" in bold style:
1. [reference/env/index.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/reference/env/index.md?plain=1#L32) - [Technical Reference - Environment Variables - Usage](https://mirrord.dev/docs/reference/env/#usage).
2. [developer/testing/index.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/developer/testing/index.md?plain=1#L142) - [Developer Guide - Testing and Development - Build mirrord-agent Docker Image](https://mirrord.dev/docs/developer/testing/#build-mirrord-agent-docker-image)
3. [reference/fileops/index.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/reference/fileops/index.md?plain=1#L136-L137) - [Technical Reference - File Operations - Syscalls - lseek](https://mirrord.dev/docs/reference/fileops/#lseek)
4. [developer/testing/index.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/developer/testing/index.md?plain=1#L228) - [Developer Guide - Testing and Development - Build and run mirrord](https://mirrord.dev/docs/developer/testing/#build-and-run-mirrord)

Quote Style:
1. [docs/overview/real-world-example.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/overview/real-world-example.md?plain=1#L63) - [Overview - Example: Debugging an Application with mirrord - Setup - Migrations](https://mirrord.dev/docs/overview/real-world-example/#migrations)
2. [docs/overview/real-world-example.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/overview/real-world-example.md?plain=1#L155) - [Overview - Example: Debugging an Application with mirrord - Outgoing Traffic](https://mirrord.dev/docs/overview/real-world-example/#outgoing-traffic)
3. [docs/overview/real-world-example.md](https://github.com/metalbear-co/mirrord.dev/blob/main/content/docs/overview/real-world-example.md?plain=1#L271) - [Overview - Example: Debugging an Application with mirrord - Traffic Steal (at the bottom of the section)](https://mirrord.dev/docs/overview/real-world-example/#traffic-steal)

Comment (t4lz) [reply content/docs/reference/traffic/index.md]: But about the content - is it better now? Or is it better just without the note?

Comment (eyalb181) [reply content/docs/reference/traffic/index.md]: Re notes - lets align everything to your style, i.e. quote + bold.
Re content, I would move the note to the bottom of the paragraph (after UDP). I'll also add some suggestions in a separate comment.

File: content/docs/reference/traffic/index.md [hunk 7]
```diff
@@ -165,9 +167,22 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
 
-mirrord's outgoing traffic feature intercepts outgoing requests from the local process and sends them through the remote pod instead.
-Responses are then routed back to the local process. A simple use case of this feature is when a user wants their local process to make
-an API call to another service in the k8s cluster, for example, a database read/write.
+mirrord's outgoing traffic feature intercepts outgoing requests from the local process and 
+sends them through the remote pod instead. Responses are then routed back to the local process.
+A simple use case of this feature is enabling the local process to make an API call to another service in the k8s
+cluster, for example, a database read/write.
+
+> **Note:** If the handling of incoming requests by your app involves outgoing API calls to other services, then it
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:175]:
```suggestion
> **Note:** If the handling of incoming requests by your app involves outgoing API calls to other services, and mirrord is configured to mirror incoming traffic, then it
```

File: content/docs/reference/traffic/index.md [hunk 8]
```diff
@@ -165,9 +167,22 @@ bigbear@metalbear:~/mirrord-demo$ curl http://192.168.49.2:32000/users
...
+> might be the case that both the version that is deployed to the cloud and the one you are running locally (which
```
Comment (eyalb181) [inline content/docs/reference/traffic/index.md:176]:
```suggestion
> might be the case that both the remote pod and the local process (which
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: content/docs/reference/traffic/index.md (hunk 4 above)
Comment (eyalb181) [reply content/docs/reference/traffic/index.md]: Also note that this is a sensitive point that comes up a lot in meetings - how do we prevent outgoing traffic from the local process from corrupting the staging environment? What you've written is sufficient for now, but in the future we'll need to suggest more solutions (e.g. directing outgoing traffic to services set up locally).

Comment (t4lz) [reply content/docs/reference/traffic/index.md]: Moved the Note to after UDP and pushed your suggestions. Also pushed a commit making a uniform style for all comments, but if you prefer I could push it in a separate PR instead.

**Review by eyalb181 — APPROVED** [review]

**Comment by eyalb181** [comment]
@t4lz LGTM, thanks!
